### PR TITLE
USWDS-Site: Added in page navigation accessibility test page

### LINKED ISSUE #2856: CC a11y: publish in-page navigation checklist - create PR
Publish the `usa-in-page-navigation` accessibility test checklist items found in the [Critical Checklist spreadsheet](https://docs.google.com/spreadsheets/d/1fAJaObQHv7aIaeUpH4fTLmauhmugMtNCrbpZe4BJcPY/edit?usp=sharing) (Google Doc :lock:). 

The process for creating a new accessibility test page can be found in the [Creating accessibility checklists - process doc](https://docs.google.com/document/d/14WTCsdzgwpoz_Miztd66RmEGfwwSEUFCVN1DkSQITTE/edit#heading=h.fa3sv1ray42g) (Google Doc :lock:).

### PR DESCRIPTION
**Pull request by RachelCorsino**
# Summary

Added accessibility test page for in page navigation component

## Related issue

Closes #2856

## Preview link
Preview link: [In-page Navigation Component Page](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/rc-in-page-nav-checklist/components/in-page-navigation/)

## Testing and review
Follow these steps:

1. Confirm that both the main component page and the accessibility tests page have the correct compliance tag at the top.
2. Confirm that the main component page links to the accessibility tests page in the side navigation.
3. Confirm that the main component page links to the accessibility tests page from the accessibility guidance section.
4. Confirm that the main component page shows the accessibility tests summary.
5. Check that accessibility tests page provides the correct counts for each test status type.
6. Confirm the test checklist summaries and data are accurate.
7. Confirm no visual issues.
8. Confirm there is an appropriate changelog entry on both the main component page and the accessibility tests page.

### CODE REVIEW

File: _components/in-page-navigation/accessibility-tests.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+permalink: /components/in-page-navigation/accessibility-tests/
+layout: accessibility-test
+component:
+ name: in page navigation
```
Comment (mejiaj) [inline _components/in-page-navigation/accessibility-tests.md:5]: Frontmatter is YAML, we do two spaces for consistent indents.

```suggestion
  name: in page navigation
```

File: _components/in-page-navigation/accessibility-tests.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+title: In page navigation accessibility tests
+category: Components
+lead: Any USWDS in page navigation component should pass these manual accessibility tests.
+changelog:
+ key: 'component-in-page-navigation-accessibility'
```
Comment (mejiaj) [inline _components/in-page-navigation/accessibility-tests.md:10]: Again, two indents here.

```suggestion
  key: 'component-in-page-navigation-accessibility'
```

File: _components/in-page-navigation/in-page-navigation.md [hunk 3]
```diff
@@ -15,26 +15,8 @@ redirect_from:
...
-- text: Accessibility
-  href: '#accessibility-in-page-navigation'
-- text: Package
-  href: '#in-page-navigation-package'
-- text: References
-  href: '#references'
-- text: Latest updates
-  href: '#changelog'
+- text: in page navigation accessibility tests
+  href: /components/in-page-navigation/accessibility-tests/
```
Comment (mejiaj) [inline _components/in-page-navigation/in-page-navigation.md:19]:
```suggestion
  - text: in page navigation accessibility tests
    href: /components/in-page-navigation/accessibility-tests/
```

File: _data/accessibility-tests/in-page-navigation.yml
Comment (mejiaj) [inline _data/accessibility-tests/in-page-navigation.yml:1]: Can you run prettier here to fix indents?

```sh
npx prettier --write '_data/accessibility-tests/in-page-navigation.yml'
```

File: _data/changelogs/component-in-page-navigation-accessibility.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+title: In page navigation accessibility tests
+type: component
+items:
+ - date: NNNN-NN-NN
+   summary: Added accessibility tests page.
+   affectsGuidance: true
+   githubPr: 2863
+   githubRepo: uswds-site
```
Comment (mejiaj) [inline _data/changelogs/component-in-page-navigation-accessibility.yml:8]:
```suggestion
title: In page navigation accessibility tests
type: component
items:
  - date: NNNN-NN-NN
    summary: Added accessibility tests page.
    affectsGuidance: true
    githubPr: 2863
    githubRepo: uswds-site

```

**Review by mejiaj — CHANGES_REQUESTED** [review]
Thanks @RachelCorsino. The pages look good and no a11y issues found.

Minor comments on code formatting below.

File: _components/in-page-navigation/accessibility-tests.md (hunk 1 above)
Comment (RachelCorsino) [reply _components/in-page-navigation/accessibility-tests.md]: Resolved in [5b4a912](https://github.com/uswds/uswds-site/pull/2863/commits/5b4a9120e93079ea8d1cc7b6e1c4a847bd6afb73)

File: _components/in-page-navigation/accessibility-tests.md (hunk 2 above)
Comment (RachelCorsino) [reply _components/in-page-navigation/accessibility-tests.md]: Resolved in [5b4a912](https://github.com/uswds/uswds-site/pull/2863/commits/5b4a9120e93079ea8d1cc7b6e1c4a847bd6afb73)

File: _data/accessibility-tests/in-page-navigation.yml
Comment (RachelCorsino) [reply _data/accessibility-tests/in-page-navigation.yml:1]: Resolved in [6037b42](https://github.com/uswds/uswds-site/pull/2863/commits/6037b429e58cd9671f70c8b6bb7f88ed610ea7ae)

**Comment by RachelCorsino** [comment]
> Hi, Rachel. I'm being super picky with this one! Can you verify that for many of these items the correct last version tested is 3.8.2? I know it's picky, but I think on 8/8 the last version was 3.8.1, right? the date for 3.8.2 is 8/9.
> 
> Also, please check the 2.4.7 criteria name(s). There are three of them, and they all say "Focus Visible," but I think(?) one is "Headings and Labels" and one is "Link Purpose." :)

Good catches @cathybaptista! These have all been resolved in [9592e28](https://github.com/uswds/uswds-site/pull/2863/commits/9592e28445fa628086f3fab23728409b88f4e772)

File: _components/in-page-navigation/accessibility-tests.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
+---
+permalink: /components/in-page-navigation/accessibility-tests/
+layout: accessibility-test
+component:
+  name: in page navigation
+title: In page navigation accessibility tests
+category: Components
+lead: Any USWDS in page navigation component should pass these manual accessibility tests.
```
Comment (mahoneycm) [inline _components/in-page-navigation/accessibility-tests.md:8]: **Chore:** All references to "in page navigation" should be hyphenated. 

```suggestion
permalink: /components/in-page-navigation/accessibility-tests/
layout: accessibility-test
component:
  name: in-page navigation
title: In-page navigation accessibility tests
category: Components
lead: Any USWDS in-page navigation component should pass these manual accessibility tests.
```

File: _components/in-page-navigation/in-page-navigation.md [hunk 6]
```diff
@@ -15,26 +15,8 @@ redirect_from:
...
-  href: '#usability-guidance-in-page-navigation'
-- text: Accessibility
-  href: '#accessibility-in-page-navigation'
-- text: Package
-  href: '#in-page-navigation-package'
-- text: References
-  href: '#references'
-- text: Latest updates
-  href: '#changelog'
+  - text: in page navigation accessibility tests
```
Comment (mahoneycm) [inline _components/in-page-navigation/in-page-navigation.md:18]: **Polish:** This navigation item should be capitalized and hyphenated.

```suggestion
  - text: In-page navigation accessibility tests
```

File: _includes/accessibility-tests/wcag-details.html [hunk 7]
```diff
@@ -122,6 +122,11 @@
   {% assign criterion_name = "Location" %}
   {% assign criterion_url = "https://www.w3.org/TR/WCAG21/#location" %}
 {% endif %}
+{% if item.wcag_criterion == "2.4.9" %}
+  {% assign criterion_level = "AAA" %}
+  {% assign criterion_name = "Location" %}
```
Comment (mahoneycm) [inline _includes/accessibility-tests/wcag-details.html:127]: **issue:** This criterion name is incorrect

```suggestion
  {% assign criterion_name = "Link purpose (link only)" %}
```

**Review by mahoneycm — CHANGES_REQUESTED** [review]
A couple small changes needed for cleanup. I only flagged one file for replacing `in page navigation` with `in-page navigation` but if you could make sure the other files match that would be great!

Let me know if you have any questions.

File: _includes/accessibility-tests/wcag-details.html (hunk 7 above)
Comment (RachelCorsino) [reply _includes/accessibility-tests/wcag-details.html]: Resolved in [5b6b02e](https://github.com/uswds/uswds-site/pull/2863/commits/5b6b02e37396e05112ca7be66889a42238fad443)

File: _components/in-page-navigation/in-page-navigation.md (hunk 6 above)
Comment (RachelCorsino) [reply _components/in-page-navigation/in-page-navigation.md]: Resolved in [5b6b02e](https://github.com/uswds/uswds-site/pull/2863/commits/5b6b02e37396e05112ca7be66889a42238fad443)

File: _data/changelogs/component-in-page-navigation-accessibility.yml [hunk 8]
```diff
@@ -0,0 +1,8 @@
+title: In-page navigation accessibility tests
+type: component
+items:
+  - date: NNNN-NN-NN
```
Comment (mejiaj) [inline _data/changelogs/component-in-page-navigation-accessibility.yml:4]: **chore:**  Will need to add date before final merge.

File: _data/changelogs/component-in-page-navigation.yml [hunk 9]
```diff
@@ -2,6 +2,11 @@ title: In-page navigation
 type: component
 changelogURL:
 items:
+  - date: NNNN-NN-NN
```
Comment (mejiaj) [inline _data/changelogs/component-in-page-navigation.yml:5]: **chore:**  Will need to add date before final merge.

**Review by mejiaj — APPROVED** [review]
Code looks good. I've made small formatting tweaks, mainly spacing and capitalized the subnav link.

**Review by alex-hull — CHANGES_REQUESTED** [review]
Criteria 2.4.8 under the general sections should be marked as conditional versus marked as pass. Basing this on previous discussion. Everything else looks good!

File: _data/accessibility-tests/in-page-navigation.yml [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+    test_status: pass
+    test_type: general
+    version_tested: 3.8.1
+    wcag_criterion: 1.4.3
+  - summary: Navigation links clearly indicate location on the website.
+    summary_additional: When you view the in-page navigation, there is distinct styling that gives clear visual indication of where you are on the page. When using a screen reader, you will hear "current" as the indicator for location.
+    test_status: pass
+    test_type: general
+    version_tested: 3.8.1
+    wcag_criterion: 2.4.8
```
Comment (mahoneycm) [inline _data/accessibility-tests/in-page-navigation.yml:28]: Related to @alex-hull's [comment](https://github.com/uswds/uswds-site/pull/2863#pullrequestreview-2373401993), we've updated this test status

>Criteria 2.4.8 under the general sections should be marked as conditional versus marked as pass. Basing this on previous discussion. Everything else looks good!

```suggestion
  - summary: Navigation links clearly indicate location on the website.
    summary_additional: When you view the in-page navigation, there is distinct styling that gives clear visual indication of where you are on the page. When using a screen reader, you will hear "current" as the indicator for location.
    test_status: conditional
    test_type: general
    version_tested: 3.8.1
    wcag_criterion: 2.4.8
```

Comment (RachelCorsino) [reply _data/accessibility-tests/in-page-navigation.yml]: Resolved in [bad51be](https://github.com/uswds/uswds-site/pull/2863/commits/bad51be98a8c4bb0366197b2eec3e597a1d341b8)

**Review by mahoneycm — APPROVED** [review]
LGTM!

**Review by alex-hull — APPROVED** [review]
Changes look good!

File: _data/changelogs/component-in-page-navigation-accessibility.yml (hunk 8 above)
Comment (thisisdano) [inline _data/changelogs/component-in-page-navigation-accessibility.yml:4]:
```suggestion
  - date: 2024-10-16
```

File: _data/changelogs/component-in-page-navigation.yml (hunk 9 above)
Comment (thisisdano) [inline _data/changelogs/component-in-page-navigation.yml:5]:
```suggestion
  - date: 2024-10-16
```

**Review by thisisdano — APPROVED** [review]

**Pull request merged**